fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -341,12 +366,28 @@ private ScoredPlan planFilter(@Nonnull PlanContext planContext, @Nonnull QueryCo
     @Nullable
     private ScoredPlan planFilter(@Nonnull PlanContext planContext, @Nonnull QueryComponent filter, boolean needOrdering) {
         final InExtractor inExtractor = new InExtractor(filter);
+        ScoredPlan withInAsOr = null;
         if (planContext.query.getSort() != null) {
-            inExtractor.setSort(planContext.query.getSort(), planContext.query.isSortReverse());
+            if (!inExtractor.setSort(planContext.query.getSort(), planContext.query.isSortReverse()) && // needs to come first, to clear sort
```

Comment:
Oh, hm. So this essentially relies on the side effect from this operation? I don't love having this happen within the if-condition (rather than it happening on the previous line and capturing the success or failure as an explicit `boolean`) as I don't think people expect side effects there. But it's probably acceptable if we want to keep it like this and take the PR prior to changing it (and the comment helps).

Reply:
I'm happy to extract this as a boolean. I don't really want to do a larger refactor, since this was definitely meant to perturb things as little as possible.

Reply:
Yeah, don't really need a larger refactor; was more just asking for it to be on its own line, though maybe that's not actually clearer.

Reply:
I think it's a little clearer.